## CI and Greptile both caught a real regression — fixed in `d3627822`

Greptile put it exactly right:

> Because synchronization runs after migration processing and no longer corrects existing records, roles automatically created from DocType permissions can retain desk access and classify their users as System Users.

And CI failed on the behavioural consequence, in all three unit-test jobs:

```
FAIL test_accept_creates_website_user_with_role
AssertionError: 'System User' != 'Website User'
```

### What I had wrong

The docstring claimed that creating roles in `after_install` "gets in first" ahead of frappe's `make_module_and_roles`. It does not. `install_app` order is:

1. `sync_for(app)` — doctype sync, and `make_module_and_roles` creates every role named in a permission with `desk_access = 1`
2. `after_install` hooks ← we run here, and a create-only pass sees the roles already exist and skips them
3. `sync_fixtures(app)` ← what the **old Role fixture** used to run in, forcing the value back to 0

Removing the fixture removed the only thing that set the install-time default. Net effect on `develop` today: a fresh Gameplan install makes every member a System User.

### The fix

Split the two responsibilities, because install and migrate genuinely own different things:

- **`setup_roles()`** — `after_install`. Forces `desk_access = 0`. Safe to write there: `frappe.flags.in_install` makes `Role.on_update` return early, and there are no users to re-evaluate yet.
- **`sync_roles()`** — `after_migrate`. Unchanged; still only creates roles that went missing, never writes to an existing one. That is the part that must stay create-only to avoid the deadlock.

### Verified both directions on gameplan-demo.test

- Set the three roles to `desk_access = 1` → `test_accept_creates_website_user_with_role` fails with the identical assertion error CI produced.
- Run `setup_roles()` under `in_install` → roles back to 0, invitation tests pass.
- Full suite: 1000 tests passing.

--- a/gameplan/roles.py
+++ b/gameplan/roles.py
@@ -47,9 +47,32 @@ def setup_roles():
 
 
 def sync_roles():
-	"""Post-migrate repair: create roles that went missing, and touch nothing else."""
-	existing = set(frappe.get_all("Role", filters={"name": ("in", GAMEPLAN_ROLES)}, pluck="name"))
+	"""Post-migrate repair: create roles that went missing, and touch nothing else.
+
+	Report — rather than fix — a role that grants desk access. Deleting a Gameplan role
+	gets it recreated by `make_module_and_roles` with `desk_access = 1` during the same
+	migrate's doctype sync, and this cannot silently correct that: the write would demote
+	users, and if it raised it would roll migrate back and retry forever. Whether a
+	Gameplan role grants desk access is the operator's call either way, so say what is
+	true and let them decide.
+	"""
+	desk_access_by_role = dict(
+		frappe.get_all(
+			"Role",
+			filters={"name": ("in", GAMEPLAN_ROLES)},
+			fields=["name", "desk_access"],
+			as_list=True,
+		)
+	)
 
 	for role_name in GAMEPLAN_ROLES:
-		if role_name not in existing:
+		if role_name not in desk_access_by_role:
 			frappe.get_doc(doctype="Role", role_name=role_name, desk_access=0).insert(ignore_permissions=True)
+
+	if granting_desk_access := sorted(
+		role for role, desk_access in desk_access_by_role.items() if desk_access
+	):
+		print(
+			f"Gameplan roles granting desk access: {', '.join(granting_desk_access)}. "
+			"Users holding them are System Users. Set desk_access = 0 if that is not intended."
+		)

--- a/gameplan/tests/platform/test_roles.py
+++ b/gameplan/tests/platform/test_roles.py
@@ -13,6 +13,8 @@ deadlock, not a one-off error — the write never lands, so the next migrate ret
 tests here pin the one rule that avoids it: create what is missing, touch nothing else.
 """
 
+import io
+from contextlib import redirect_stdout
 from unittest.mock import patch
 
 import frappe
@@ -64,7 +66,7 @@ class TestGameplanRoles(FrappeTestCase):
 		"""
 		frappe.get_doc(doctype="Role", role_name=TEST_ROLE, desk_access=1).insert()
 
-		with patch("gameplan.roles.GAMEPLAN_ROLES", (TEST_ROLE,)):
+		with patch("gameplan.roles.GAMEPLAN_ROLES", (TEST_ROLE,)), redirect_stdout(io.StringIO()):
 			sync_roles()
 
 		self.assertEqual(frappe.db.get_value("Role", TEST_ROLE, "desk_access"), 1)
@@ -77,12 +79,32 @@ class TestGameplanRoles(FrappeTestCase):
 		"""
 		frappe.get_doc(doctype="Role", role_name=TEST_ROLE, desk_access=1).insert()
 
-		with patch.object(Role, "on_update", autospec=True) as on_update:
+		with patch.object(Role, "on_update", autospec=True) as on_update, redirect_stdout(io.StringIO()):
 			sync_roles()
 			with patch("gameplan.roles.GAMEPLAN_ROLES", (TEST_ROLE,)):
 				sync_roles()
 
 		on_update.assert_not_called()
+
+	def test_a_role_granting_desk_access_is_reported_not_corrected(self):
+		"""Deleting a Gameplan role gets it recreated by make_module_and_roles with
+		desk_access = 1 during the same migrate. We cannot fix that silently, so it has to
+		at least show up in the migrate output."""
+		frappe.get_doc(doctype="Role", role_name=TEST_ROLE, desk_access=1).insert()
+
+		output = io.StringIO()
+		with patch("gameplan.roles.GAMEPLAN_ROLES", (TEST_ROLE,)), redirect_stdout(output):
+			sync_roles()
+
+		self.assertIn(TEST_ROLE, output.getvalue())
+		self.assertEqual(frappe.db.get_value("Role", TEST_ROLE, "desk_access"), 1)
+
+	def test_nothing_is_reported_when_no_role_grants_desk_access(self):
+		output = io.StringIO()
+		with patch("gameplan.roles.GAMEPLAN_ROLES", (TEST_ROLE,)), redirect_stdout(output):
+			sync_roles()
+
+		self.assertEqual(output.getvalue(), "")
 
 	def test_all_gameplan_roles_exist(self):
 		for role in GAMEPLAN_ROLES:

